indra/newview/llvoavatar.cpp

```diff
@@ -10763,17 +10763,18 @@ void LLVOAvatar::updateImpostors()
     LLViewerCamera::sCurCameraID = LLViewerCamera::CAMERA_WORLD;
 
     std::vector<LLCharacter*> instances_copy = LLCharacter::sInstances;
-    for (std::vector<LLCharacter*>::iterator iter = instances_copy.begin();
-        iter != instances_copy.end(); ++iter)
+    for (LLCharacter* instance : instances_copy)
     {
-        LLVOAvatar* avatar = (LLVOAvatar*) *iter;
-        if (!avatar->isDead()
-            && avatar->isVisible()
-            && avatar->isImpostor()
-            && avatar->needsImpostorUpdate())
+        if (LLVOAvatar* avatar = dynamic_cast<LLVOAvatar*>(instance))
         {
-            avatar->calcMutedAVColor();
-            gPipeline.generateImpostor(avatar);
```

Comment:
What is this going to change crash wise? This doesn't looke like it will change anything at all and shouldn't be accosiated with a crash fix.

Reply:
The `dynamic_cast` ensures type safety here.

Reply:
It does not, LLVOAvatar is the only use for LLAvatarAppearance/LLCharacter

Reply:
> What is this going to change crash wise?

This is just a kind off null-check

Reply:
sInstances can not have null inside, it gets cleared in LLCharacter's destructor so it is pointless to do a null check.

Reply:
What could be a reason for the signal SIGSEGV on the line 10616?
![image](https://github.com/user-attachments/assets/4485b0d9-35e1-43e9-b773-5a14d0bb4254)
![image](https://github.com/user-attachments/assets/ee09711e-8c1c-4fe4-a672-8e456a0deaf5)
I see the only explanation that `avatar` is NULL (`avatar->mDrawable.mPointer` is being checked)
Your thoughts?

Reply:
There's nothing wrong with a null check. But the question is, why is it a nullptr at this point?

Reply:
> P.S. other places also don't do null checks for sInstances, sInstances is intended to be safe in this regard.

It seems there are some critical places where this check is being done
![image](https://github.com/user-attachments/assets/d768a779-e636-4a52-bbe1-3cbe2f943ac9)
![image](https://github.com/user-attachments/assets/0c81399b-895a-42b6-883c-50227cf64594)
![image](https://github.com/user-attachments/assets/173ad2e3-7edf-40d9-b53a-49b58a195864)

Reply:
> What could be a reason for the signal SIGSEGV on the line 10616?
Invalid pointer, perhaps you should make sure characters aren't created/destroyed outside of main thread?